client/engine/chainservice/eth_chainservice.go

```diff
@@ -111,25 +116,41 @@ func (ecs *EthChainService) SendTransaction(tx protocols.ChainTransaction) error
 }
 
 func (ecs *EthChainService) subcribeToEvents() error {
+
+	go ecs.listenForLogEvents()
+	return nil
+}
+
+func (ecs *EthChainService) listenForLogEvents() {
```

Comment:
I am probably missing something, but why is the below logic moved to `listenForLogEvents`?

Reply:
> Do you think our issue and [ethereum/go-ethereum#23845](https://github.com/ethereum/go-ethereum/issues/23845) are related? The geth issue description talks about missing events after days of running a script, not minutes...

Based on our test run behaviour I'm pretty sure that our issue is being caused by the [geth issue](https://github.com/ethereum/go-ethereum/issues/23845) as:
- Without this change (with a long running test) we see the withdraw transaction submitted but no chain events in the logs
- Without this change tests will fail if they run over ~3 minutes


>In general, having to resubscribe to events every minute seems a bit wild.

Agreed 😔. 

>Also, is it possible to miss events while a subscription is inactive (after cancellation but before resubscription)?

That's a good point! I haven't run into that problem when running testground runs. I think we may be ok but I'll investigate.

Reply:
> Based on our test run behaviour I'm pretty sure that our issue is being caused by the [geth issue](https://github.com/ethereum/go-ethereum/issues/23845) as:
> 
> * Without this change (with a long running test) we see the withdraw transaction submitted but no chain events in the logs
> * Without this change tests will fail if they run over ~3 minutes

For a test with a long runtime, does the test open/close more on-chain channels than a short running test? Or do longer running tests open/use/close just more virtual channels?

We use a websocket for connecting to an [Ethereum JSON rpc endpoint](https://github.com/statechannels/go-nitro-testground/blob/a0a5a200487d36fb1e62e057ef9212f55e67c79b/chain/chain-service.go#L22). I wonder if there is some timeout issues with that websocket...

Reply:
>For a test with a long runtime, does the test open/close more on-chain channels than a short running test? Or do longer running tests open/use/close just more virtual channels?

Nope, the long running test would create the same amount of ledger channels, so we should have the same amount of on chain channels (every participant creates a ledger channel with every hub).

Reply:
> I am probably missing something, but why is the below logic moved to `listenForLogEvents`?

Sorry I accidentally replied to this comment for another issue 🙃.  Since `listenForLogEvents` is now periodically recreating `sub` every 60 seconds I though it made more sense for `listenForLogEvents` to create the `sub` the first time`. This required moving the logic into `listenForLogEvents`. Otherwise `listenForLogEvents` would accept a `sub` that it overwrites in 60 seconds.

Reply:
I did a little spelunking through `go-ethereum` and found [this](https://github.com/ethereum/go-ethereum/blob/e14164d516600e9ac66f9060892e078f5c076229/eth/filters/filter_system.go#L43):

```
// Config represents the configuration of the filter system.
type Config struct {
	LogCacheSize int           // maximum number of cached blocks (default: 32)
	Timeout      time.Duration // how long filters stay active (default: 5min)
}
```

5 minutes seems roughly in line with the failure's we've seen, so that might explain it.

Reply:
Does it makes sense to change our resubscription period to something like 4 minutes?

Reply:
> Does it makes sense to change our resubscription period to something like 4 minutes?

Yup, now that we know it's a 5 minute timeout it makes sense to use a larger value.